pkg/webhook/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -65,6 +75,40 @@ func checkRevisionName(appConfig *v1alpha2.ApplicationConfiguration) (bool, stri
 	return true, ""
 }
 
+// checkWorkloadNameForVersioning check whether versioning-enabled component workload name is empty
+func checkWorkloadNameForVersioning(ctx context.Context, client client.Reader, appConfig *v1alpha2.ApplicationConfiguration) (bool, string) {
+	for _, v := range appConfig.Spec.Components {
+		acc := v
+		vEnabled, err := checkComponentVersionEnabled(ctx, client, &acc)
+		if err != nil {
+			return false, fmt.Sprintf(errFmtCheckWorkloadName, err.Error())
+		}
+		if vEnabled {
+			// only check versioning-enabled compoents
+
+			//TODO(roywang) share almost the same render functions with AppConfig controller
```

Comment:
yep, we should not copy the controller logic here

Reply:
Currently, those logics are still private in appConfig controller e.g. Render.go. I'm not sure whether it's allowed for me to make them public... If it's okay, I would like to make them public or move them into common OAM lib (e.g. util/helper.go) and reuse them here.

Reply:
Make them public if there's a need.

Reply:
Got it. I will fix.